src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +13,400 @@
+        newQnForm.setQuestionNumberSuffix("");
+        
+        newQnForm.setQuestionTypeOptions(getQuestionTypeChoiceOptions());
+      
+        FeedbackQuestionGeneralSettings generalSettings = new FeedbackQuestionGeneralSettings();  
+        newQnForm.setGeneralSettings(generalSettings);
+        generalSettings.setGiverParticipantOptions(getParticipantOptions(null, true));
+        generalSettings.setRecipientParticipantOptions(getParticipantOptions(null, false));
+        generalSettings.setNumOfEntitiesToGiveFeedbackToValue(1);
+        
+        newQnForm.setNumOfQuestionsOnPage(questions.size());
+        newQnForm.setQuestionNumberOptions(getQuestionNumberOptions(questions.size() + 1));
+      
+        newQnForm.setQuestionSpecificEditFormHtml(getNewQuestionSpecificEditFormHtml());
+        newQnForm.setEditable(true);
+        
+        setDefaultVisibilityOptions(generalSettings);
+    }
+
+
+    private void setDefaultVisibilityOptions(FeedbackQuestionGeneralSettings generalSettings) {
+        Map<String, Boolean> isGiverNameVisible = new HashMap<String, Boolean>();
+        Map<String, Boolean> isRecipientNameVisible = new HashMap<String, Boolean>();
+        Map<String, Boolean> isResponsesVisible = new HashMap<String, Boolean>();
+        
+        List<FeedbackParticipantType> participantVisiblity = new ArrayList<FeedbackParticipantType>();
+        participantVisiblity.add(FeedbackParticipantType.INSTRUCTORS);
+        participantVisiblity.add(FeedbackParticipantType.RECEIVER);
+        
+        for (FeedbackParticipantType participant : participantVisiblity) {
+            isGiverNameVisible.put(participant.name(), true);
+            isRecipientNameVisible.put(participant.name(), true);
+            isResponsesVisible.put(participant.name(), true);
+        }
+        
+        generalSettings.setGiverNameVisibleFor(isGiverNameVisible);
+        generalSettings.setRecipientNameVisibleFor(isRecipientNameVisible);
+        generalSettings.setResponseVisibleFor(isResponsesVisible);
+    }
+    
+    
+    private void buildBasicFsForm(FeedbackSessionAttributes newFeedbackSession) {
+        fsForm = new FeedbackSessionsForm();
+        
+        fsForm.setFsDeleteLink(new Url(getInstructorFeedbackSessionDeleteLink(newFeedbackSession.courseId, newFeedbackSession.feedbackSessionName, "")));
+        fsForm.setCopyToLink(new Url(getFeedbackSessionEditCopyLink()));
+        
+        fsForm.setCourseIdForNewSession(newFeedbackSession.courseId);
+        
+        fsForm.setFsNameEditable(false);
+        fsForm.setFsName(newFeedbackSession.feedbackSessionName);
+        
+        fsForm.setCourseIdEditable(false);
+        fsForm.setCourses(null);
+        
+        fsForm.setEditFsButtonsVisible(true);
+        fsForm.setFeedbackSessionTypeEditable(false);
+        fsForm.setFeedbackSessionTypeOptions(null);
+
+        fsForm.setFeedbackSessionNameForSessionList(null);
+        
+        fsForm.setCoursesSelectField(null);
```

Comment:
Do setting these as `null` actually result in anything?
If the default constructor for `fsForm` does not set them to anything else, wouldn't this just set `null` objects to `null`?